int8 Xr
int8 Yr
